smartsim/_core/entrypoints/telemetrymonitor.py

```diff
@@ -0,0 +1,676 @@
+                    entity.name, entity.step_id, entity.step_id, entity.is_managed
+                )
+            self._tracked_runs[run.timestamp] = run
+
+    def on_modified(self, event: FileModifiedEvent) -> None:
+        """Event handler for when a file or directory is modified.
+
+        :param event: Event representing file/directory modification.
+        :type event: FileModifiedEvent"""
+        super().on_modified(event)  # type: ignore
+        self.process_manifest(event.src_path)
+
+    def on_created(self, event: FileCreatedEvent) -> None:
+        """Event handler for when a file or directory is created.
+
+        :param event: Event representing file/directory creation.
+        :type event: FileCreatedEvent"""
+        super().on_created(event)  # type: ignore
+        self.process_manifest(event.src_path)
+
+    def _to_completed(
+        self,
+        timestamp: int,
+        entity: JobEntity,
+        experiment_dir: pathlib.Path,
+        step_info: t.Optional[StepInfo],
+    ) -> None:
+        """Move a monitored entity from the active to completed collection to
+        stop monitoring for updates during timesteps.
+
+        :param timestamp: the current timestamp for event logging
+        :type timestamp: int
+        :param entity: the running SmartSim Job
+        :type entity: JobEntity
+        :param experiment_dir: the experiement directory to monitor for changes
+        :type experiment_dir: pathlib.Path
+        :param entity: the StepInfo received when requesting a Job status update
+        :type entity: t.Optional[StepInfo]
+        """
+        inactive_entity = self._tracked_jobs.pop(entity.key)
+        if entity.key not in self._completed_jobs:
+            self._completed_jobs[entity.key] = inactive_entity
+
+        job = self.job_manager[entity.name]
+        self.job_manager.move_to_completed(job)
+
+        if step_info:
+            detail = f"status: {step_info.status}, error: {step_info.error}"
+        else:
+            detail = "unknown status. step_info not retrieved"
+
+        track_event(
+            timestamp,
+            entity.name,
+            entity.job_id,
+            entity.step_id,
+            entity.type,
+            "stop",
+            experiment_dir,
+            self._logger,
+            detail=detail,
+        )
+
+    def on_timestep(self, timestamp: int, experiment_dir: pathlib.Path) -> None:
+        """Called at polling frequency to request status updates on
+        monitored entities
+
+        :param timestamp: the current timestamp for event logging
+        :type timestamp: int
+        :param experiment_dir: the experiement directory to monitor for changes
+        :type experiment_dir: pathlib.Path"""
+        launcher = self.launcher
+        entity_map = self._tracked_jobs
+
+        names = {entity.name: entity for entity in entity_map.values()}
+
+        if launcher and names:
```

Comment:
No need to check against launcher here, it will always come back non-nullable

Reply:
removed extraneous conditional